src/components/Modal/src/ModalLayer.vue

```diff
@@ -155,6 +162,16 @@ export default {
 	destroyed() {
 		this.unwatchStackedModal();
 	},
+
+	methods: {
+		closeOnOutsideClick(event) {
+			const { closeOnOutsideClick } = this.currentLayer.state.options;
+			const { modal } = this.$refs;
+			if (modal && closeOnOutsideClick && !modal.contains(event.target)) {
+				this.modalApi.close();
```

Comment:
@pretzelhammer expressed concern that this implementation should support close interception (eg. confirmation dialog before close).

IMO, this implementation will allow for that because it closes the modal via `close()`, and interception should intercept all calls to `close()`.

So, if we were to implement something like a `before-close` hook, the specific modal instance can pass in a hook:
```html
<o-modal :before-close="beforeCloseHook">
```

where
```js
async beforeCloseHook() {
   // anything async
  return true; // or false to block
}
```

the `close()` method will be updated to something like:
```js
async close() {
  if (typeof this.options.beforeCloseHook === 'function') {
    if (!(await this.options.beforeCloseHook())) {
      return; // cancel
    }
  }

  // close modal
}
```

Reply:
@Ryguy11 could you add @privatenumber 's suggestion to this PR? I think it would be nice if we had this functionality and knew it worked before settling on an implementation.